app/models/user.rb

```diff
@@ -4,20 +4,30 @@ class User < ApplicationRecord
   # Connects this user object to Curation Concerns behaviors.
   include CurationConcerns::User
 
+  include Hydra::RoleManagement::UserRoles
+
   if Blacklight::Utils.needs_attr_accessible?
     attr_accessible :email, :password, :password_confirmation
   end
   # Connects this user object to Blacklights Bookmarks.
   include Blacklight::User
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+  devise :omniauthable
 
   # Method added by Blacklight; Blacklight uses #to_s on your
   # user class to get a user-displayable login/identifier for
   # the account.
   def to_s
     email
   end
+
+  def self.from_omniauth(access_token)
+    User.where(provider: access_token.provider, uid: access_token.uid).first_or_create do |user|
+      user.uid = access_token.uid
+      user.provider = access_token.provider
+      user.username = access_token.uid
```

Comment:
A password field is required to be stored in the db otherwise sign in doesn't work for "new" users: https://github.com/pulibrary/pulmap/blob/master/app/models/user.rb#L23